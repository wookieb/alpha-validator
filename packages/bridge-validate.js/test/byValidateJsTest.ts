import {byValidateJs} from "@src/index";

describe('byValidateJs', () => {
    const SCHEMA = 'foo';

    it('no violations', () => {
        return expect(
            byValidateJs({
                name: {
                    presence: true,
                    length: {
                        minimum: 5
                    }
                }
            })({
                name: 'wookieb'
            }, SCHEMA)
        )
            .resolves
            .toBeUndefined();
    });

    it('simple validation', () => {
        return expect(
            byValidateJs({
                name: {
                    presence: true,
                    length: {
                        minimum: 5
                    }
                }
            })({
                name: 'foo'
            }, SCHEMA)
        )
            .resolves
            .toMatchSnapshot();
    });

    it('nested validation', () => {
        return expect(
            byValidateJs({
                'profile.name': {
                    presence: true,
                    length: {
                        minimum: 5
                    }
                }
            })({
                profile: {
                    name: 'foo'
                }
            }, SCHEMA)
        )
            .resolves
            .toMatchSnapshot();
    })
});