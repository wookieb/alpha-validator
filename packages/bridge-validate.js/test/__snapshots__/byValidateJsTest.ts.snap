// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`byValidateJs nested validation 1`] = `
ViolationsList {
  "violations": [
    {
      "message": "Profile name is too short (minimum is 5 characters)",
      "path": [
        "profile",
        "name",
      ],
    },
  ],
}
`;

exports[`byValidateJs simple validation 1`] = `
ViolationsList {
  "violations": [
    {
      "message": "Name is too short (minimum is 5 characters)",
      "path": [
        "name",
      ],
    },
  ],
}
`;
